backend/src/middleware/handleHtmlContent/handleContentData.js

```diff
@@ -1,39 +1,46 @@
 import extractMentionedUsers from './notifications/extractMentionedUsers'
 import extractHashtags from './hashtags/extractHashtags'
 
-const notify = async (postId, idsOfMentionedUsers, context) => {
+const notifyMentions = async (label, id, idsOfMentionedUsers, context) => {
+  if (!idsOfMentionedUsers.length) return
+
   const session = context.driver.session()
   const createdAt = new Date().toISOString()
   const cypher = `
-    MATCH(p:Post {id: $postId})<-[:WROTE]-(author:User)
-    MATCH(u:User)
+    MATCH (source)
+    WHERE source.id = $id AND $label IN LABELS(source)
+    MATCH (source)<-[:WROTE]-(author: User)
+    MATCH (u: User)
     WHERE u.id in $idsOfMentionedUsers
     AND NOT (u)<-[:BLOCKED]-(author)
```

Comment:
The `author` must be different here. Consider the following situation:

1. (personA)-[:BLOCKED]->(personB)
2. (personA)-[:WROTE]->(post)
3. (personC)-[:WROTE]->(comment)-[:COMMENTED]->(post)

Now what happens if `personC` would mention `personB` in post authored by `personA`? As `personB` you would get a notification linked to a post and following it you get a 404. **That's** the situation we want to avoid.

It is no problem right now if the author of a comment mentions a blocked user. The blocked user could still follow the link and not get a 404. We would change this behaviour as soon as your blocked users cannot see your comments anymore. But I don't think that we will implement this behaviour, because you would have totally different conversation flows per user account.

Reply:
Could you write a backend test for this? Have a look into [this example here](https://github.com/Human-Connection/Human-Connection/blob/master/backend/src/middleware/handleHtmlContent/handleContentData.spec.js#L170).

Reply:
Done 😆